packages/glimmer-runtime/tests/in-element-test.ts

```diff
@@ -0,0 +1,414 @@
+  MyRootView[CLASS_META].seal();
+
+  env.begin();
+  view.appendTo('#qunit-fixture');
+  env.commit();
+
+  return view;
+}
+
+QUnit.module('Targeting a remote element', {
+  setup() {
+    env = new TestEnvironment();
+  }
+});
+
+QUnit.test('basic', function(assert) {
+  let externalElement = document.createElement('div');
+
+  appendViewFor(
+    stripTight`{{#-in-element externalElement}}[{{foo}}]{{/-in-element}}`,
+    { externalElement, foo: 'Yippie!' }
+  );
+
+  equalsElement(externalElement, 'div', {}, stripTight`[Yippie!]`);
+
+  set(view, 'foo', 'Double Yips!');
+  rerender();
+
+  equalsElement(externalElement, 'div', {}, stripTight`[Double Yips!]`);
+
+  set(view, 'foo', 'Yippie!');
+  rerender();
+
+  equalsElement(externalElement, 'div', {}, stripTight`[Yippie!]`);
+});
+
+QUnit.test('changing to falsey', function(assert) {
+  let first = document.createElement('div');
+  let second = document.createElement('div');
+
+  appendViewFor(
+    stripTight`
+      |{{foo}}|
+      {{#-in-element first}}[{{foo}}]{{/-in-element}}
+      {{#-in-element second}}[{{foo}}]{{/-in-element}}
+    `,
+    { first, second: null, foo: 'Yippie!' }
+  );
+
+  equalsElement(first, 'div', {}, `[Yippie!]`);
+  equalsElement(second, 'div', {}, ``);
+  assertAppended('|Yippie!|<!----><!---->');
+
+  set(view, 'foo', 'Double Yips!');
+  rerender();
+
+  equalsElement(first, 'div', {}, `[Double Yips!]`);
+  equalsElement(second, 'div', {}, ``);
+  assertAppended('|Double Yips!|<!----><!---->');
+
+  set(view, 'first', null);
+  rerender();
+
+  equalsElement(first, 'div', {}, ``);
+  equalsElement(second, 'div', {}, ``);
+  assertAppended('|Double Yips!|<!----><!---->');
+
+  set(view, 'second', second);
+  rerender();
+
+  equalsElement(first, 'div', {}, ``);
+  equalsElement(second, 'div', {}, `[Double Yips!]`);
+  assertAppended('|Double Yips!|<!----><!---->');
+
+  set(view, 'foo', 'Yippie!');
+  rerender();
+
+  equalsElement(first, 'div', {}, ``);
+  equalsElement(second, 'div', {}, `[Yippie!]`);
+  assertAppended('|Yippie!|<!----><!---->');
+
+  set(view, 'first', first);
+  set(view, 'second', null);
+  rerender();
+
+  equalsElement(first, 'div', {}, `[Yippie!]`);
+  equalsElement(second, 'div', {}, ``);
+  assertAppended('|Yippie!|<!----><!---->');
+});
+
+QUnit.test('with pre-existing content', function(assert) {
+  let externalElement = document.createElement('div');
+  let initialContent = externalElement.innerHTML = '<p>Hello there!</p>';
+
+  appendViewFor(
+    stripTight`{{#-in-element externalElement}}[{{foo}}]{{/-in-element}}`,
+    { externalElement, foo: 'Yippie!' }
+  );
+
+  equalsElement(externalElement, 'div', {}, `${initialContent}[Yippie!]`);
+
+  set(view, 'foo', 'Double Yips!');
+  rerender();
+
+  equalsElement(externalElement, 'div', {}, `${initialContent}[Double Yips!]`);
+
+  set(view, 'foo', 'Yippie!');
+  rerender();
+
+  equalsElement(externalElement, 'div', {}, `${initialContent}[Yippie!]`);
+});
```

Comment:
Can we test setting the element to `null` as part of the steps here to ensure we aren't removing the initial content?